### PR TITLE
[Profile] az login: Add back the warning that a browser is opened

### PR DESCRIPTION
**Pull request by jiasli**
**Description**  

Revert the change in #10851. The removal of the warning message that a browser is opened leads to confusions: https://github.com/MicrosoftDocs/azure-docs-cli/issues/2093, #14473.

If the browser fails to open (due to some unknown reasons), Azure CLI will hang forever and the user won't know what to do next. 

```
> az login
_
```
With this PR, the warning is shown again:

```
> az login
The default web browser has been opened at https://login.microsoftonline.com/common/oauth2/authorize. Please continue the login in the web browser.
If no web browser is available or if the web browser fails to open, use device code flow with `az login --use-device-code`.
```

**Testing Guide**  
```
az login
```

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S173

**Review by yonzhan — APPROVED** [review]
LGTM

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 1]
```diff
@@ -1272,6 +1272,12 @@ def log_message(self, format, *args):  # pylint: disable=redefined-builtin,unuse
         results['no_browser'] = True
         return
 
+    # Emit a warning to inform that a browser is opened.
+    # Only show the path part of the URL and hide the query string.
+    logger.warning("The default web browser has been opened at %s. Please continue the login in the web browser.\n"
+                   "If no web browser is available or if the web browser fails to open, use device code flow "
```
Comment (evelyn-ys) [inline src/azure-cli-core/azure/cli/core/_profile.py:1278]: If the web browser does fail to open, will CLI run into this line or hang over line 1269 `succ = open_page_in_browser(url)` ?

**Review by evelyn-ys — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/_profile.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/_profile.py:1278]: If the web browser fails to open, `open_page_in_browser` should return `False`. This warning is for the scenario where `open_page_in_browser` returned `True`, but the browser is not brought to the foreground for some reason. (This mainly happens on some Linux distributions or MacOS.)

**Pull request merged**